src/utils/RewardsSweeper.sol

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.28;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { IAccountingModule } from "../AccountingModule.sol";
+import { Initializable } from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import { AccessControlUpgradeable } from "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+import { IERC4626 } from "@openzeppelin/contracts/interfaces/IERC4626.sol";
+
+/**
+ * @title RewardsSweeper
+ * @notice Contract to sweep rewards from a strategy and process them through the accounting module
+ */
+contract RewardsSweeper is Initializable, AccessControlUpgradeable {
+    using SafeERC20 for IERC20;
+
+    bytes32 public constant REWARDS_SWEEPER_ROLE = keccak256("REWARDS_SWEEPER_ROLE");
+
+    IAccountingModule public accountingModule;
+
+    error Unauthorized();
+    error TransferFailed();
+    error CannotSweepRewards();
+
+    event RewardsSwept(uint256 amount);
+    event AccountingModuleUpdated(address newModule, address oldModule);
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    /**
+     * @notice Initializes the contract
+     * @param admin The address of the admin
+     * @param accountingModule_ The address of the accounting module
+     */
+    function initialize(address admin, address accountingModule_) external initializer {
+        __AccessControl_init();
+        _grantRole(DEFAULT_ADMIN_ROLE, admin);
+
+        accountingModule = IAccountingModule(accountingModule_);
+    }
+
+    function sweepRewardsUpToAPRMax() public onlyRole(REWARDS_SWEEPER_ROLE) {
+        sweepRewardsUpToAPRMax(accountingModule.snapshotsLength() - 1);
+    }
+
+    function sweepRewardsUpToAPRMax(uint256 snapshotIndex) public onlyRole(REWARDS_SWEEPER_ROLE) {
+        // Calculate max rewards based on current TVL and target APY
+        uint256 totalAssets = IERC4626(accountingModule.STRATEGY()).totalAssets();
+
+        uint256 timeElapsed = block.timestamp - accountingModule.snapshots(snapshotIndex).timestamp;
```

Comment:
providing a `snapshotIndex` higher than the latest added snapshot will result in `.timestamp` returning `0`, resulting in `timeElapsed` being equal to `block.timestamp`, resulting in `maxRewards` being extremely high.

add:

```solidity
if (snapshotIndex >= accountingModule.snapshotsLength()) revert InvalidSnapshot(snapshotIndex, accountingModule.snapshotsLength() - 1);
// where the 2nd arg is the max available snapshotIndex
```